opacus/optimizers/ddp_perlayeroptimizer.py

```diff
@@ -92,12 +92,14 @@ def _ddp_per_layer_hook(
 
         return p.grad
 
-    def register_hooks(self, module: GradSampleModule):
-        module.ddp_hooks = []
+    def register_hooks(self):
         for p, max_grad_norm in zip(self.params, self.max_grad_norms):
             if not p.requires_grad:
                 continue
 
-            module.ddp_hooks.append(
+            if not hasattr(p, "ddp_hooks"):
```

Comment:
ha, they actually are related to PrivacyEngine.

Before these changes `DistributedPerLayerOptimizer` required special handling in `make_private` method: 

```
optimizer.register_hook(model)
```

This call was unique for DistributedPerLayerOptimizer because of the optimization that only kicks in when both distributed training and per layer clipping is enabled. (see #196)

With these changes we can ditch an extra call (move it to optimizer's `__init__`) and provide unified `make_private` implementation for all optimizers